crates/component-macro/src/lib.rs

```diff
@@ -884,7 +884,8 @@ impl Expander for ComponentTypeExpander {
                 const SIZE32: usize = {
                     let mut size = 0;
                     #sizes
-                    #internal::align_to(#discriminant_size as usize, Self::ALIGN32) + size
+                    #internal::align_to(#discriminant_size as usize, Self::ALIGN32)
+                    + #internal::align_to(size, Self::ALIGN32)
```

Comment:
Could this be updated to match the [canonical definition](https://github.com/WebAssembly/component-model/blob/main/design/mvp/CanonicalABI.md#size) which aligns-up the summation of the discriminant and the max variant size rather than aligning-up each individual summation? I think it probably ends up being the same thing but figure it may be best to align (heh) with the spec where we can.